bindings/node/native/src/lib.rs

```diff
@@ -22,102 +21,68 @@ use tokio::runtime::Runtime;
-            let message = event.cloned_message();
-            let confirmed = *event.confirmed();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                if let Some(message) = account.messages_mut().iter_mut().find(|m| m == &&message) {
-                    message.set_confirmed(Some(confirmed));
-                }
-            });
-        });
-        iota_wallet::event::on_reattachment(|event| {
-            let message = event.cloned_message();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                account.append_messages(vec![message]);
-            });
-        });
-        iota_wallet::event::on_broadcast(|event| {
-            let message = event.cloned_message();
-            mutate_account_if_exists(event.account_id(), move |account| {
-                if let Some(message) = account.messages_mut().iter_mut().find(|m| m == &&message) {
-                    message.set_broadcasted(true);
-                }
-            });
-        });
-        Default::default()
-    });
+fn account_instances() -> &'static AccountInstanceMap {
+    static INSTANCES: Lazy<AccountInstanceMap> = Lazy::new(Default::default);
     &INSTANCES
 }
 
-pub(crate) fn get_account(id: &str) -> Arc<RwLock<Account>> {
-    let map = instances()
+pub(crate) fn get_account(id: &AccountIdentifier) -> AccountGuard {
+    let map = account_instances()
         .read()
         .expect("failed to lock account instances: get_account()");
     map.get(id).expect("account dropped or not initialised").clone()
 }
 
-pub(crate) fn store_account(account: Account) -> String {
-    let mut map = instances()
+pub(crate) fn store_account(account: AccountGuard) -> AccountIdentifier {
+    let mut map = account_instances()
         .write()
         .expect("failed to lock account instances: store_account()");
-    let id: String = thread_rng().sample_iter(&Alphanumeric).take(10).collect();
-    map.insert(id.clone(), Arc::new(RwLock::new(account)));
+    let id = {
+        let account_ = account.read().unwrap();
+        account_.id().clone()
+    };
+    map.insert(id.clone(), account);
     id
 }
 
-pub(crate) fn update_account(id: &str, account: Account) {
-    let mut map = instances()
+pub(crate) fn remove_account(id: &AccountIdentifier) {
+    let mut map = account_instances()
         .write()
-        .expect("failed to lock account instances: store_account()");
-    map.insert(id.to_string(), Arc::new(RwLock::new(account)));
+        .expect("failed to lock account instances: remove_account()");
+    map.remove(id);
+}
+
+/// Gets the synced account instances map.
+fn synced_account_instances() -> &'static SyncedAccountInstanceMap {
+    static INSTANCES: Lazy<SyncedAccountInstanceMap> = Lazy::new(Default::default);
+    &INSTANCES
 }
 
-pub(crate) fn remove_account(id: &str) {
-    let mut map = instances()
+pub(crate) fn get_synced_account(id: &str) -> SyncedAccountGuard {
```

Comment:
probably already on the todo list: error handling and return a result

Reply:
in this case it's ok because a panic shouldn't happen here